src/Bicep.Core/Registry/AzureContainerRegistryManager.cs

```diff
@@ -152,11 +153,13 @@ private static async Task<OciArtifactResult> DownloadManifestAsync(IOciArtifactR
             ValidateManifestResponse(manifestResponse);
 
             var deserializedManifest = OciManifest.FromBinaryData(manifestResponse.Value.Manifest) ?? throw new InvalidOperationException("the manifest is not a valid OCI manifest");
-            var layers = deserializedManifest.Layers
-                .Select(layer => new KeyValuePair<string, BinaryData>(layer.Digest, PullLayerAsync(client, layer).Result))
-                .ToImmutableDictionary(pair => pair.Key, pair => pair.Value);
+            var layers = new List<(string MediaType, BinaryData Data)>();
+            foreach (var layer in deserializedManifest.Layers)
+            {
+                layers.Add((layer.MediaType, await PullLayerAsync(client, layer)));
+            }
```

Comment:
Can you explain the reasoning behind indexing the layers by MediaType? it assumes that no two layers can have the same MediaType which is not a constraint of an ACR artifact.

Reply:
Funny you say that, your (I think) code hαd a dictionary indexed by media type. I changed that to a list to avoid exactly this. Mine is just a list of tuples containing the media type for convenience so caller doesn't have to interpret the binary layer because generally they will be selecting media type but allows multiple layers with same type.

Reply:
It was indexing by digest as far as I remember